js2/webhosting/lesson.md

```diff
@@ -0,0 +1,133 @@
+Zatím jsme všechny aplikace zkoušeli jen u sebe na počítači.¨
+V prohlížeči píšeme do daresního řádku `localhost`, což označuje tento (místní) počítač – ten počítač, na kterém adresu napíšu.
+Kdybyste se chtěly svými stránkami pochlubit a někomu poslaly odkaz na `localhost`, pokusí se prohlížeč připojit na jeho lokální počítač – kde ale vaše aplikace neběží.
+
+My ale chceme vytvářet www stránky, kde to „www“ je zkratka z „World Wide Web“, tedy celosvětová síť.
+Potřebujeme tedy, aby naše webové stránky byly dostupné z celého světa, ne jen z našeho počítače.
+
+Zpřístupnit náš počítač celému světu by bylo technicky obtížné a nepohodlné – asi byste nechtěli, aby váš notebook musel pořád běžet a být připojený k internetu, protože co kdyby se někdo chtěl podívat na váš web.
+Řešením je umístit naše webové stránky u někoho, kdo má trvale běžící server, který je trvale připojený do internetu (a má spolehlivější a rychlejší připojení k internetu, než my; má také zálohované napájení elektrickým proudem).
+Dotyčný (a jeho server) bude náš web *hostovat*, proto se té službě říká **webhosting**.
+
+Než se pustíme do toho, jaký webhosting vybrat a jak s ním pracovat, musíme udělat odbočku k tomu, jakému typu webhostingu se vyhnout.
+Webhosting je totiž už hodně stará záležitost, pochází ještě z doby, kdy se webové stránky psaly bez JavaScriptu a často i bez CSS, žádný React neexistoval, a webové aplikace se psaly nejčastěji v PHP.
+Pokud se dnes řekne *webhosting* obvykle se tím myslí tento starý typ webhostingu.
+A tomu my se chceme vyhnout, protože poskytuje služby, které nechceme (třeba právě PHP), ale museli bychom za ně platit.
+Naopak tento typ hostingu neposkytuje služby, které nám usnadní práci – třeba spolupráci s Gitem a GitHubem.
+Pokud tedy narazíte na webhosting jako Wedos, Active24, Savana nebo jiný webhosting, který se bude chlubit tím, které verze PHP podporuje, vyhněte se mu.
+Nemusí být špatný pro PHP, ale není to nic pro naše aplikace.
+Zprovoznit je na tomto typu hostingu by sice šlo, ale bylo by to zbytečně pracné.
+
+My budeme hledat služby, které ještě nemají svůj ustálený název – někdy se jim říká třeba cloud hosting, serverless deployment platform nebo třeba [Jamstack](https://jamstack.org) deployment.
+Nejznámějšími zástupci těchto služeb jsou:
+
+* [Netlify](https://www.netlify.com)
+* [Vercel](https://vercel.com/) (dříve se jmenoval Zeit.co, kdybyste na to někde narazily)
```

Comment:
Zeit se snad nikde neobjevuje. Podle mě je to zbytečná informace navíc.

Reply:
Pořád ještě na název Zeit.co v různých materiálech na webu narážím…